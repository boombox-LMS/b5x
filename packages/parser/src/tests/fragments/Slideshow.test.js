const {
  exampleSlideshowMarkup,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/Slideshow");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Slideshow fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleSlideshowMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed slideshow fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed slideshow fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
