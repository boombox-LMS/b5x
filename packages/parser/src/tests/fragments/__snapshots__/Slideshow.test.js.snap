// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Slideshow fragment builds as expected The parsed slideshow fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
    "test-fragment-uri",
    "test-fragment-uri",
  ],
  "contentType": "Slideshow",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "b8868ef3c196f3eb247b8bb3c447f749",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
